handlers/tasks.go

```diff
@@ -0,0 +1,318 @@
+		return
+	}
+
+	// Получаем id вставленной задачи
+	id, err := res.LastInsertId()
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`{"error": "Ошибка при получении ID задачи: %v"}`, err), http.StatusInternalServerError)
+		return
+	}
+
+	// Возвращаем id добавленной задачи
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(map[string]interface{}{"id": id})
+}
+
+func putTask(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+	var task models.Task
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		http.Error(w, `{"error": "Ошибка десериализации JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Title == "" {
+		http.Error(w, `{"error": "Не указан заголовок задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Comment == "" {
+		http.Error(w, `{"error": "Не указан коммент задачи"}`, http.StatusBadRequest)
+		return
+	}
+	if task.Repeat == "" {
+		http.Error(w, `{"error": "Не указан повторение задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Date == "" || task.Date == "today" {
+		task.Date = time.Now().Format("20060102")
+	}
+
+	// Добавляем отладочные сообщения
+	fmt.Println("Редактирование задачи. Дата задачи:", task.Date)
+	fmt.Println("Заголовок задачи:", task.Title)
+	fmt.Println("Комментарий задачи:", task.Comment)
+	fmt.Println("Правило повторения:", task.Repeat)
+
+	// Проверяем, существует ли задача с данным id
+	query := "SELECT id FROM scheduler WHERE id = ?"
+	var existingID int
+	err = db.QueryRow(query, task.ID).Scan(&existingID)
+	if err != nil || existingID == 0 {
+		http.Error(w, `{"error": "Задача не найдена"}`, http.StatusNotFound)
+		return
+	}
+
+	// Обновляем задачу в базе данных
+	updateQuery := "UPDATE scheduler SET date = ?, title = ?, comment = ?, repeat = ? WHERE id = ?"
+	_, err = db.Exec(updateQuery, task.Date, task.Title, task.Comment, task.Repeat, task.ID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`{"error": "Ошибка при обновлении задачи: %v"}`, err), http.StatusInternalServerError)
+		return
+	}
+
+	// Возвращаем пустой JSON в случае успеха
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(map[string]interface{}{})
+}
+func deleteTask(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+	// Получаем идентификатор задачи из параметров запроса.
+	taskID := r.URL.Query().Get("id")
+	if taskID == "" {
+		// Если ID не передан, возвращаем ошибку.
+		http.Error(w, `{"error":"missing task ID"}`, http.StatusBadRequest)
+		return
+	}
+
+	// Пробуем удалить задачу из базы данных.
+	result, err := db.Exec("DELETE FROM scheduler WHERE id = ?", taskID)
+	if err != nil {
+		// Если возникла ошибка при выполнении запроса, возвращаем ошибку сервера.
+		http.Error(w, `{"error": "Ошибка при удалении задачи"}`, http.StatusInternalServerError)
```

Comment:
Теряешь сообщение с ошибкой, нужно либо в лог его выводить, либо клиенту выводить

Как пример
```suggestion
		http.Error(w, fmt.Sprintf(`{"error": "Ошибка при удалении задачи: %v"}`, err) http.StatusInternalServerError)
```